swarm.go

```diff
@@ -304,6 +317,31 @@ func parseEnsAPIAddress(s string) (tld, endpoint string, addr common.Address) {
 	return
 }
 
+// parseRnsAPIAddress parses string according to format
+// [contract-addr@]url and returns RNSClientConfig structure
+// with endpoint and contract address
+func parseRnsAPIAddress(s string) (endpoint string, addr common.Address) {
```

Comment:
As far as I can see, this function is exactly the same as parseEnsAPIAddress. I think it would be better to refactor those to avoid code multiplication

Reply:
I agree there is code duplication.
Fixed refactored for parseResolverAPIAddress